src/module3/routers/login.ts

```diff
@@ -0,0 +1,25 @@
+import { Router, Request, Response, NextFunction } from "express";
+import { JWT_SECRET } from "../config";
+import { loginSchema } from "../validation";
+import { validateSchema, BadRequestError } from "../utils";
+import { User } from "../data-access";
+
+const jwt = require('jsonwebtoken');
+
+const loginRouter = Router();
+const userDto = new User();
+
+loginRouter.post('/login', validateSchema(loginSchema), async (req: Request, res: Response, next: NextFunction) => {
+    try {
+        const user = await userDto.getUserByLogin(req.body);
+        if(user) {
+            const token = jwt.sign(user.toJSON(), JWT_SECRET, {expiresIn: 120});
```

Comment:
If you are providing expiry, it is a good idea to refresh the token so that the user doesn't need to sign in again and again and is provided with a different valid token.